volume/local/local.go

```diff
@@ -90,10 +91,13 @@ func New(scope string, rootUID, rootGID int) (*Root, error) {
 		r.volumes[name] = v
 		optsFilePath := filepath.Join(rootDirectory, name, "opts.json")
 		if b, err := ioutil.ReadFile(optsFilePath); err == nil {
-			v.opts = &optsConfig{}
-			if err := json.Unmarshal(b, v.opts); err != nil {
+			opts := optsConfig{}
+			if err := json.Unmarshal(b, &opts); err != nil {
 				return nil, err
 			}
+			if !reflect.DeepEqual(opts, optsConfig{}) {
```

Comment:
nit: would be good to add a comment that this is here for backwards compatibility with buggy versions.